instrumentation.go

```diff
@@ -463,3 +474,12 @@ func WithResourceAttributes(attrs ...attribute.KeyValue) InstrumentationOption {
 		return c, nil
 	})
 }
+
+// WithLogLevel returns an [InstrumentationOption] that will configure
+// an [Instrumentation] with the logger level visibility defined as inputed.
+func WithLogLevel(level string) InstrumentationOption {
```

Comment:
Instead of accepting a string with special values, we should introduce a `LogLevel` type with `enum` values. We might want to use the [OTel levels](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-severitynumber). For example:

https://github.com/open-telemetry/opentelemetry-go/blob/main/log/severity.go

Reply:
Do you think we need to create something which uses numbers as reference like the one you linked?

Reply:
That seems reasonable to me.

Reply:
We already do something like this for offsetgen:

https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/e85ddd9de6c5829d6736d7ccefbc3733f2cb8ed8/internal/tools/inspect/cmd/offsetgen/main.go#L59

We could provided human readable flags for the CLI though (i.e. `--log-level info`).

Reply:
Yeah, I usually prefer something human readable to CLI because is more quick to identify.

Reply:
@MrAlias I have create a simple Level Enum for logs, what do you think?

Reply:
> @MrAlias I have create a simple Level Enum for logs, what do you think?

This still accepts a `string`.

```suggestion
func WithLogLevel(level Level) InstrumentationOption {
```

Reply:
@MrAlias  This case here expect an string because is the input when use flag or from env, so for properly work is these scenarios this need to be string

Reply:
The environment variable needs to be parsed before it is passed to this function then.

We need to use syntax here to ensure the best possible chance values passed will be valid. Using an arbitrary string with magic values is not appropriate.

Reply:
@MrAlias changed `WithLogLevel` to work on the behaviour you said